os/arch/arm/src/amebasmart/amebasmart_pmhelpers.c

```diff
@@ -116,29 +116,15 @@ int SOCPS_AONWakeReason(void)
 void pg_timer_int_handler(void *Data)
 {
 	pmvdbg("PM Timer interrupt handler!!\n");
-	switch (g_pm_timer.timer_type) {
-		case PM_LOCK_TIMER:
-			// Relax the lock on the PM state transitions after the timer interrupt
-			pm_relax(PM_IDLE_DOMAIN, PM_NORMAL);
-			break;
-		case PM_WAKEUP_TIMER:
-			// Switch status back to normal mode after wake up from interrupt
-			pm_activity(PM_IDLE_DOMAIN, 9);
-			break;
-		default:
-			pmdbg("Timer callback triggered without setting timer, Unexpected Error!!!\n");
-			break;
-	}
-	// Reset the global struct
-	g_pm_timer.timer_type = PM_NO_TIMER;
-	g_pm_timer.timer_interval = 0;
+	// Switch status back to normal mode after wake up from interrupt
+	pm_activity(PM_IDLE_DOMAIN, 9);		
 }
 
-void up_set_pm_timer(void) {
+void up_set_pm_timer(unsigned int timer_interval) {
```

Comment:
[up_set_pm_timer](https://github.com/Samsung/TizenRT/blob/46fb6e5bd9fc0f2138be0982a8100ae82b9ab061/os/arch/arm/src/amebasmart/amebasmart_idle.c#L95)

we don't need to change this api??